src/main/java/com/sandogh/sandogh/home/HomePageController.java

```diff
@@ -1,17 +1,117 @@
 package com.sandogh.sandogh.home;
 
+import com.sandogh.sandogh.base.utils.PasswordEncryptionUtils;
+import com.sandogh.sandogh.users.UserDTO;
+import com.sandogh.sandogh.users.dao.UserDAO;
+import com.sandogh.sandogh.users.entity.User;
+import com.sandogh.sandogh.users.exceptions.UsernameAlreadyTakeException;
+import com.sandogh.sandogh.users.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import javax.validation.Valid;
 
 /**
  * @author Ehsan Soleimani (esoleimani@voipfuture.com)
  **/
 @Controller
 public class HomePageController {
+    @Autowired
+    UserService userService;
+    @Autowired
```

Comment:
Write your code so that never have dependencies on DAOs in Controller classes.
This dependency should be removed.

Reply:
Done